app/code/community/Bolt/Boltpay/Helper/Api.php

```diff
@@ -165,121 +165,141 @@ public function verify_hook($payload, $hmacHeader)
-                    'rates' => $this->getRatesDebuggingData($rates),
-                    'service' => $service,
-                    'shipping_address' => var_export($shippingAddress->debug(), true)
+            // setting Bolt as payment method
+            $immutableQuote->getShippingAddress()->setPaymentMethod(Bolt_Boltpay_Model_Payment::METHOD_CODE)->save();
+            $payment = $immutableQuote->getPayment();
+            $payment->setMethod(Bolt_Boltpay_Model_Payment::METHOD_CODE);
+
+            // adding transaction data to payment instance
+            $payment->setAdditionalInformation('bolt_transaction_status', $transactionStatus);
+            $payment->setAdditionalInformation('bolt_reference', $reference);
+            $payment->setAdditionalInformation('bolt_merchant_transaction_id', $transaction->id);
+            //add if called by ajax request, this info would be used for log
+            $payment->setAdditionalInformation('is_ajax_request', $isAjaxRequest);
+            $payment->setTransactionId($transaction->id);
+
+            Mage::helper('boltpay')->collectTotals($immutableQuote, true)->save();
+
+            // a call to internal Magento service for order creation
+            $service = Mage::getModel('sales/service_quote', $immutableQuote);
+
+            try {
+                $service->submitAll();
+            } catch (Exception $e) {
+                Mage::helper('boltpay/bugsnag')->addBreadcrumb(
+                    array(
+                        'transaction'   => json_encode((array)$transaction),
+                        'quote_address' => var_export($immutableQuote->getShippingAddress()->debug(), true)
+                    )
                 );
-                Mage::helper('boltpay/bugsnag')->notifyException(new Exception($errorMessage), $metaData);
+                throw $e;
             }
-        }
-
-        // setting Bolt as payment method
-        $immutableQuote->getShippingAddress()->setPaymentMethod(Bolt_Boltpay_Model_Payment::METHOD_CODE)->save();
-        $payment = $immutableQuote->getPayment();
-        $payment->setMethod(Bolt_Boltpay_Model_Payment::METHOD_CODE);
 
-        // adding transaction data to payment instance
-        $payment->setAdditionalInformation('bolt_transaction_status', $transactionStatus);
-        $payment->setAdditionalInformation('bolt_reference', $reference);
-        $payment->setAdditionalInformation('bolt_merchant_transaction_id', $transaction->id);
-        //add if called by ajax request, this info would be used for log
-        $payment->setAdditionalInformation('is_ajax_request', $isAjaxRequest);
-        $payment->setTransactionId($transaction->id);
-
-        Mage::helper('boltpay')->collectTotals($immutableQuote, true)->save();
-      
-        // a call to internal Magento service for order creation
-        $service = Mage::getModel('sales/service_quote', $immutableQuote);
+        } catch ( Exception $e ) {
```

Comment:
should these try block cover until like 355? Maybe we want to re-activate quote when exception happens at `Mage::getModel('boltpay/payment')->handleOrderUpdate($order)` for instance.

Reply:
No.  The code is correct as is.  If there is an exception prior to 297, it means the order was never created in Magento and the web hooks should be free to try to create it.

If there is an exception after that point, it means the order is already in Magento.  Web hooks should not be able to create a new order.